### PR TITLE
add support for taking multiple notes

### PR DESCRIPTION
**Pull request by sandiz**
Hi plibither8,
great work on the extension! like the idea of maintaining notes in the browser since its always open and available. The only thing i felt was missing was multiple notes support!

This pull request adds support for taking multiple notes. I have kept the current functionality intact and added support for notes via `notes` object in localStorage. The `Default` note data is read from `localStorage.{rawText | history | lastUpdated }`, for all others its read from `localStorage.notes.note_id.{rawText | history | lastUpdated }`

[Screenshot](https://www.screencast.com/t/x04vbd3I1)

Please take a look and let me know what you think!

### CODE REVIEW

**Comment by sandiz** [comment]
demo: https://sandiz.surge.sh/

**Comment by plibither8** [comment]
Hi sandiz,

This is an idea I hadn't thought of at all and your implementation is beautiful! It fits great into the design and does not hinder the minimalist-feel of the editor. I will look over each of the additions carefully when I get time later in the day. I'll merge the changes when I'm sure.

One of the issues I can see is that the "create-note" icon on the bottom bar is colorful. I personally would prefer a black-and-white icon that shows a "list". Secondly, users are free to create notes that have coinciding/repeating names. I think it's best if we add a simple check to prevent that.

Finally, after you've done a major chunk of work on the JS file, run `npm run lint` and `npm run lint:fix` in the command line. It'll help to pass the Travis CI tests and bring uniformity.

Apart from those and other minor style changes (which I can do), I think your work is great! Please pull changes from the repo as I've updated the `index.js` file with fresh commits fixing other issues. Thank you so much :smile:, looking forward to the updates.

**Comment by plibither8** [comment]
How are these: https://www.flaticon.com/search?word=list&order_by=1&color=1 ?

**Comment by sandiz** [comment]
@plibither8  all done with the changes, please take a look

Added
1. duplicate note detection
2. monochrome svg for notes
3. support cursorLastPosition for non-default notes

**Comment by plibither8** [comment]
Hey @sandiz, I apologise for not merging this PR, to be honest I had forgotten about it. In your latest commits, there are a few merge conflicts that are being caused because your local repo is not even with the latest few commits on the master branch. Please pull those commits and apply your changes on top of them. Thanks!

File: src/styl/index.styl [hunk 1]
```diff
@@ -278,4 +322,54 @@ section.settings
...
+                background none
+
+section.notes-add, section.notes-edit
+    height 25vh
+    top 40%
+    width 50%
+
+.note-add-input
+    height: 30px;
+    width: 193px;
```
Comment (aslafy-z) [inline src/styl/index.styl]: I'd vote to replace this hardcoded width with `100%` and add a `flex-grow: 1` to `textfield flex > .note-add-input` to let the input grow.

File: src/js/index.js [hunk 2]
```diff
@@ -68,29 +82,56 @@ const toggleDisplay = (n) => {
  * line instead of the last line so that it is visible
  * From https://stackoverflow.com/a/8190890/
  */
-const moveCaretToStart = () => {
+const moveCaretToStart = (selectionEnd = -1, selectionStart = -1) => {
 
     if (typeof textarea.selectionStart === 'number') {
         textarea.selectionStart = textarea.selectionEnd = 0;
+        if(selectionEnd !== -1)
```
Comment (aslafy-z) [inline src/js/index.js]: Code is not linted: `if(`

File: src/js/index.js [hunk 3]
```diff
@@ -534,12 +684,18 @@ const initiate = () => {
...
+    getHtmlElement('#edit')             .addEventListener('click', () => { edit();                                         }, false);
+    getHtmlElement('#save')             .addEventListener('click', () => { save();                                         }, false);
+    getHtmlElement('#lastEdited')       .addEventListener('click', () => { openModal(historySection, populateHistoryHtml); }, false);
+    getHtmlElement('#closeHistory')     .addEventListener('click', () => { closeModal(historySection);                     }, false);
+    getHtmlElement('#settings')         .addEventListener('click', () => { openModal(settingsSection, settingsControl);    }, false);
+    getHtmlElement('#closeSettings')    .addEventListener('click', () => { closeModal(settingsSection);                    }, false);
+    getHtmlElement('#settings-add')     .addEventListener('click', () => { openModal(notelistSection);                     }, false);
+    getHtmlElement('#closeNotesAdd')    .addEventListener('click', () => { closeModal(noteaddSection);                     }, false);
```
Comment (aslafy-z) [inline src/js/index.js]: It would be best if the input could be focussed after the modal has been opened.

**Pull request closed without merging**